### PR TITLE
Multi-Threading modelcompiler

### PR DESCRIPTION
**Pull request by fluffyfreak**
# Description:
So a while ago @HybridDog asked if we could use threads to speed up the modelcompiler #3782 and I said no...

...so here's a PR where I have threaded the modelcompiler!

# Of Note:
I'm not 100% sure that this is a great idea, I mean it is faster, *MUCH* faster on my 12-thread PC. However it does introduce a mutex into the texture cache that will not currently be useful in Pioneer.

On the plus side, it's one more thing out of the way of us eventually multithreading the loading, and other sections of the game engine thus speeding that stuff up.

In short, have a look, tell me what you think people.

### CODE REVIEW

**Comment by fluffyfreak** [comment]
@impaktor / @laarmen any comment?

**Comment by laarmen** [comment]
Not yet, maybe tomorrow.

Le sam. 24 sept. 2016 14:04, Andrew Copland <notifications@github.com> a
écrit :

**Comment by impaktor** [comment]
I'm not against this, I just don't see the speed of modelcompiler as an issue. But if it can spin-off to a future where pioneer itself can run multithreaded, then why not.

**Comment by fluffyfreak** [comment]
Ok, i've made it a little more formal, with a static `Init` method for the TextureBuilder rather than relying on static initialisation ordering by the compiler :)

Will merge when the checks have completed.
